src/biz/metrics.go

```diff
@@ -95,15 +109,31 @@ func (mc *MetricsFetcher) Process(containerID string) (Sample, error) {
 	if err != nil {
 		return metrics, err
 	}
-	// TODO: move logic to skip container without State to Docker specific code.
 	if json.ContainerJSONBase == nil {
 		return metrics, errors.New("empty container inspect result")
 	}
 
+	// TODO: move logic to skip container without State to Docker specific code.
 	if json.State == nil {
 		log.Debug("invalid container %s JSON: missing State", containerID)
 	}
 
+	if mc.exitedContainerTTL != 0 && strings.ToLower(json.State.Status) == "exited" {
```

Comment:
if json.State == nil this part should be skipped to avoid a nil ptr error right?